pkg/storage/endpoint/gc_states.go

```diff
@@ -120,6 +120,79 @@ var (
 	_ json.Unmarshaler = (*ServiceSafePoint)(nil)
 )
 
+// TimeOptional is time.Time or nil
+type TimeOptional struct {
+	*time.Time
+}
+
+func (t *TimeOptional) int64() int64 {
+	if t.Time != nil {
+		return t.UnixNano()
+	}
+	return 0
+}
+
+// MarshalJSON implements json.Marshaler for TimeOptional
+func (t *TimeOptional) MarshalJSON() ([]byte, error) {
+	return json.Marshal(t.int64())
+}
+
+// UnmarshalJSON implements json.Marshaler for TimeOptional
+func (t *TimeOptional) UnmarshalJSON(b []byte) error {
+	var expireAt int64
+	if err := json.Unmarshal(b, &expireAt); err != nil {
+		return err
+	}
+	if expireAt < math.MaxInt64 && expireAt > 0 {
+		val := time.Unix(0, expireAt)
+		t.Time = &val
+	} else {
+		t.Time = nil
+	}
+	return nil
+}
+
+// GlobalGCBarrier represents a global GC barrier.
+// It looks like a GCBarrier now but the struct might change with the code evolve.
+// A more important reason is to distinguish from GC barrier by the type system,
+// avoiding potential misuse between them in code.
+type GlobalGCBarrier struct {
+	BarrierID string       `json:"barrier_id"`
+	BarrierTS uint64       `json:"barrier_ts"`
+	ExpiredAt TimeOptional `json:"expired_at"`
```

Comment:
The service safe points introduces many negative examples of namings. `ExpiredAt` looks like describing something that has already been expired, and records a past time at which the expiration happened.
As part of the new-designed GC, please keep the naming style consistent with `GCBarrier`, instead of `ServiceGCSafePoint`.